src/SFA.DAS.PR.Jobs/Services/RelationshipService.cs

```diff
@@ -0,0 +1,168 @@
+﻿using Microsoft.Extensions.Logging;
+using SFA.DAS.PR.Data;
+using SFA.DAS.PR.Data.Common;
+using SFA.DAS.PR.Data.Entities;
+using SFA.DAS.PR.Data.Repositories;
+
+namespace SFA.DAS.PR.Jobs.Services;
+
+public interface IRelationshipService
+{
+    Task CreateRelationship<T>(ILogger<T> _logger, RelationshipModel relationshipModel, CancellationToken cancellationToken) where T : class;
+}
+
+public record struct RelationshipModel(
+    long? AccountLegalEntityId,
+    string? AccountLegalEntityPublicHashedId,
+    long ProviderUkprn,
+    string? AccountPublicHashId,
+    string NotificationTemplateName,
+    string permissionAuditAction
+);
+
+public sealed class RelationshipService(
+    IAccountLegalEntityRepository _accountLegalEntityRepository,
+    IProvidersRepository _providersRepository,
+    IAccountProviderRepository _accountProviderRepository,
+    IProviderRelationshipsDataContext _providerRelationshipsDataContext,
+    IAccountProviderLegalEntityRepository _accountProviderLegalEntityRepository,
+    IPermissionAuditRepository _permissionAuditRepository
+) : IRelationshipService
+{
+    private const string SystemUser = "System";
+
+    public async Task CreateRelationship<T>(ILogger<T> _logger, RelationshipModel relationshipModel, CancellationToken cancellationToken) where T : class
```

Comment:
Instead of using model with nullable properties, can we have two overloads, it will require breaking down this method in to smaller private units which each overload will invoke as needed?